### PR TITLE
Refactor mock function replacement logic to fix issues related to callOriginal cases

### LINKED ISSUE #35764: Address the issues in #35535 related to function mocking callOriginal case
**Description:**
Due the changes done in the PR https://github.com/ballerina-platform/ballerina-lang/pull/35535, an issue arises with the ordering of the test execution. 
eg:
```ballerina
test:when(mock_intAdd).thenReturn(100);
test:when(mock_stringAdd).callOriginal();
test:assertEquals(intAdd(10, 6), 100);
test:assertEquals(stringAdd("hello"), "hi there!");
```
Above case is not working as expected due to the changes introduced in the above PR.

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #35764

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/testerina-integration-test/src/test/resources/project-based-tests/function-mocking-tests/tests/main_test.bal [hunk 1]
```diff
@@ -239,6 +239,14 @@ public function callOriginal_Test1() {
     test:assertEquals(floatAdd(10.6, 4.5), 15.1);
 }
 
+@test:Config {}
+public function callOriginal_Test2() {
```
Comment (Dilhasha) [inline tests/testerina-integration-test/src/test/resources/project-based-tests/function-mocking-tests/tests/main_test.bal:243]: @gabilang I don't see a test case with `thenReturn` followed by `callOriginal` for the same function within a test case. Shall we add that too?

Comment (gabilang) [reply tests/testerina-integration-test/src/test/resources/project-based-tests/function-mocking-tests/tests/main_test.bal:243]: For the same function it always going to take the last mock return type right? And it didn't give any issue with previous logic too. The issue observed with different functions when they are used to assert something like follows:
```
test:when(mock_intAdd).thenReturn(100);
test:when(mock_stringAdd).callOriginal();
test:assertEquals(intAdd(10, 6), 100);
test:assertEquals(stringAdd("hello"), "default");
```

Comment (Dilhasha) [reply tests/testerina-integration-test/src/test/resources/project-based-tests/function-mocking-tests/tests/main_test.bal:243]: Yeah you are right. But we had a gap of such a test case here and it's good to have. Thanks for adding :)

File: misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/MockFunctionReplaceVisitor.java [hunk 2]
```diff
@@ -61,41 +58,23 @@ public MockFunctionReplaceVisitor(int api, ClassWriter cw, String name, String m
...
-                super.visitMethod(access, name, desc, signature, exceptions), mockFunc);
-    }
+            defaultMethodVisitor = super.visitMethod(access, name, desc, signature, exceptions);
+        } else {
+            // Rename function as $ORIG_name.
+            defaultMethodVisitor = super.visitMethod(access, ORIGINAL_FUNC_NAME_PREFIX + name,
+                    desc, signature, exceptions);
 
+            // Create a new method with the same name of original method and call the method $MOCK_name inside it.
+            MethodVisitor mv = super.visitMethod(access, name, desc, signature, exceptions);
```
Comment (warunalakshitha) [inline misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/MockFunctionReplaceVisitor.java:71]: Shall we move duplicate method creation logic tos eperate method

Comment (gabilang) [reply misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/MockFunctionReplaceVisitor.java]: Addressed with the commit https://github.com/ballerina-platform/ballerina-lang/pull/36115/commits/d7ac7b336c1cc3f8d8eba078c20f21ebc279088c

**Pull request merged**